### PR TITLE
Refactor Stage to be an enum (Closes #1877)

### PR DESCRIPTION
**Pull request by drivenflywheel**
Replaces #338

### CODE REVIEW

**Review by andrewbp — DISMISSED** [review]

File: src/main/java/emissary/core/MobileAgent.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public abstract class MobileAgent implements IMobileAgent, MobileAgentMBean {
     protected transient Thread thread = null;
 
     // Name for our threads
-    public static final String AGENT_THREAD = "MobileAgent-".intern();
```
Comment (drivenflywheel) [inline src/main/java/emissary/core/MobileAgent.java:43]: String literals are always interned, so these are redundant calls.  (Not technically within the scope of the original request, but these aren't numerous).

**Review by fbruton — DISMISSED** [review]

**Review by ldhardy — CHANGES_REQUESTED** [review]
rebase please

**Comment by ldhardy** [comment]
> > rebase please
> 
> FYI I think any member of the repo can click the rebase button. Click the dropdown next to 'Update Branch' and 'Update with Rebase'

I wasn't sure if that was a passive aggressive move or not.  Culturally.  I'm willing to try it so I can give feedback as long as it's not perceived as guache.

**Comment by drivenflywheel** [comment]
> I wasn't sure if that was a passive aggressive move or not. Culturally. I'm willing to try it so I can give feedback as long as it's not perceived as gauche.

You assume we know what "gauche" means. 😄

**Review by ldhardy — APPROVED** [review]
👍🏽

**Review by andrewbp — APPROVED** [review]

**Comment by jpdahlke** [comment]
This should integrate with no changes downstream.

**Pull request merged**